Tests/BSGConfigurationBuilderTests.m

```diff
@@ -0,0 +1,494 @@
+    BugsnagConfiguration *config =
+        [BSGConfigurationBuilder configurationFromOptions:@{
+            @"endpoints" : @{
+                @"notify" : @"https://notify.example.com",
+                @"sessions" : @"https://sessions.example.com"
+            },
+            @"apiKey" : @"b128acce"
+        }];
+    XCTAssertNotNil(config);
+    XCTAssertEqualObjects(@"b128acce", config.apiKey);
+    XCTAssertEqualObjects([NSURL URLWithString:@"https://notify.example.com"],
+                          config.notifyURL);
+    XCTAssertEqualObjects([NSURL URLWithString:@"https://sessions.example.com"],
+                          config.sessionURL);
+}
+
+- (void)testDecodeReleaseStageWithoutApiKey {
+    BugsnagConfiguration *config = [BSGConfigurationBuilder configurationFromOptions:@{
+        @"releaseStage" : @[ @"three" ],
+    }];
+    XCTAssertNil(config);
+}
+
+- (void)testDecodeReleaseStageInvalidType {
+    BugsnagConfiguration *config = [BSGConfigurationBuilder configurationFromOptions:@{
+        @"releaseStage" : @NO,
+        @"apiKey" : @"ac928faeec"
+    }];
+    XCTAssertNil(config);
+}
+
+- (void)testDecodeReleaseStage {
+    BugsnagConfiguration *config =
+        [BSGConfigurationBuilder configurationFromOptions:@{
+            @"releaseStage" : @"debug",
+            @"apiKey" : @"b128acce"
+        }];
+    XCTAssertNotNil(config);
+    XCTAssertEqualObjects(@"b128acce", config.apiKey);
+    XCTAssertEqualObjects(@"debug", config.releaseStage);
+}
+
+- (void)testDecodeNotifyReleaseStagesInvalidTypes {
+    BugsnagConfiguration *config = [BSGConfigurationBuilder configurationFromOptions:@{
+        @"notifyReleaseStages" : @[ @"beta", @"prod", @300 ],
+        @"apiKey" : @"5628acc"
+    }];
+    XCTAssertNil(config);
+    config = [BSGConfigurationBuilder configurationFromOptions:@{
+        @"notifyReleaseStages" : @{@"name" : @"foo"},
+        @"apiKey" : @"5628acc"
+    }];
+    XCTAssertNil(config);
+    config = [BSGConfigurationBuilder configurationFromOptions:@{
+        @"notifyReleaseStages" : @"fooo",
+        @"apiKey" : @"5628acc"
+    }];
+    XCTAssertNil(config);
+}
+
+- (void)testDecodeNotifyReleaseStagesWithoutApiKey {
+    BugsnagConfiguration *config = [BSGConfigurationBuilder configurationFromOptions:@{
+        @"notifyReleaseStages" : @[ @"beta", @"prod" ],
+    }];
+    XCTAssertNil(config);
+}
+
+- (void)testDecodeNotifyReleaseStages {
+    BugsnagConfiguration *config =
+        [BSGConfigurationBuilder configurationFromOptions:@{
+            @"notifyReleaseStages" : @[ @"beta", @"prod" ],
+            @"apiKey" : @"b128acce"
+        }];
+    XCTAssertNotNil(config);
+    XCTAssertEqualObjects(@"b128acce", config.apiKey);
+    XCTAssertEqual(2, config.notifyReleaseStages.count);
+    XCTAssertTrue([config.notifyReleaseStages containsObject:@"beta"]);
+    XCTAssertTrue([config.notifyReleaseStages containsObject:@"prod"]);
+}
+
+- (void)testDecodeReportOOMsWithoutApiKey {
```

Comment:
It seems like there only needs to be 1 test case in this suite to check what happens if an API key isn't present